File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int ans = 0;
        int prev = 0;

        for(auto it = bank.begin(); it != bank.end(); ++it) {
            int count = 0;
            for(auto jt = it->begin(); jt != it->end(); ++jt) {
                if(*jt == '1') {
                    count++;
                }
            }
            if(count != 0) {
                ans += prev*count;
                prev = count;
            }
        }
        return ans;
    }
};

File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.go
func numberOfBeams(bank []string) int {
    prev := 0
    ans := 0

    for _, val := range(bank) {
        count := 0
        for _,inner_val := range(val) {
            if inner_val == '1' {
                count++
            }
        }
        if count != 0 {
            ans += prev*count
            prev = count
        }
    }   
    return ans
}